index.js

```diff
@@ -1,15 +1,29 @@
 'use strict'
+/**
+ * @typedef {import('unified').FrozenProcessor} FrozenProcessor
+ * @typedef {import('unified').RunCallback} RunCallback
+ * @typedef {import('unified').Settings} Settings
+ * @typedef {import('unified').Transformer} Transformer
+ * @typedef {import('unist').Node} Node
+ */
 
 var hast2mdast = require('hast-util-to-mdast')
 
 module.exports = attacher
 
-// Attacher.
-// If a destination is given, runs the destination with the new mdast tree
-// (bridge-mode).
-// Without destination, returns the mdast tree: further plugins run on that tree
-// (mutate-mode).
+/**
+ * Attacher.
+ *
+ * If a destination is given, runs the destination with the new mdast
+ * tree (bridge-mode). Without destination, returns the mdast tree: further
+ * plugins run on that tree (mutate-mode).
+ *
+ * @param {FrozenProcessor} destination
+ * @param {Settings} options
+ * @returns {Transformer}
+ */
 function attacher(destination, options) {
+  /** @type {Partial<FrozenProcessor> & {document?: boolean | null}} */
```

Comment:
the overload noted above may remove the need for this?
If not, it would likely be
```suggestion
  /** @type {FrozenProcessor | Settings | undefined} */
```
or
```suggestion
  /** @type {FrozenProcessor | Settings} */
```

Reply:
Thanks. I did something here...